test/testdata/infer/lub_and_glb_corner_cases.rb

```diff
@@ -51,4 +51,27 @@ def generic_class_glb_collapse
   def generic_class_module_glb
     T.assert_type!(0, T.all(A, T::Hash[String, String])) # error: does not have asserted type `T.all(T::Hash[String, String], A)`
   end
+
+  def generic_class_lub(b)
+    if b == 1
+      xs = T::Array[T.nilable(String)].new
+    elsif b == 2
+      xs = nil
+    else
+      xs = T::Array[T.untyped].new
+    end
+    
+    # T.any(nil, T::Array[T.nilable(String)], T::Array[T.untyped]) => T.nlable(T::Array[T.untyped])
+    T.assert_type!(xs, T.nilable(T::Array[T.untyped]))
```

Comment:
Could you use `T.reveal_type` here instead?

Reply:
Talked in DM; `reveal_type` shows the actual type Sorbet sees and doesn't reduce it, so it's better for testing lub. `T.assert_type!` calls into lub and may reduce the type further.